connector_spec/connector_spec_without_comments.ts

```diff
@@ -0,0 +1,348 @@
+            z.object({
+                type: z.literal("crt"),
+                cert: z.string().describe("Base64-encoded certificate"),
+                key: z.string().describe("Base64-encoded private key"),
+                passphrase: z.string().optional(),
+                ca: z.string().optional().describe("Base64-encoded CA certificate"),
+                verificationMode: z.enum(["none", "certificate", "full"]).optional(),
+            }),
+            z.object({
+                type: z.literal("pfx"),
+                pfx: z.string().describe("Base64-encoded PFX bundle"),
+                passphrase: z.string().optional(),
+                ca: z.string().optional().describe("Base64-encoded CA certificate"),
+                verificationMode: z.enum(["none", "certificate", "full"]).optional(),
+            }),
+        ]),
+    }),
+    z.object({
+        method: z.literal("aws_sig_v4"),
+        credentials: z.object({
+            accessKey: z.string(),
+            secretKey: z.string(),
+            region: z.string().optional(),
+            service: z.string().optional(),
+        }),
+    }),
+    z.object({
+        method: z.literal("webhook"),
+        url: z.string().url(),
+        extraHeaders: z.record(z.string(), z.string()).optional(),
+    }),
+]);
+
+export type AuthConfig = z.infer<typeof AuthSchema>;
+
+export interface RateLimitPolicy {
+    strategy: "header" | "status_code" | "response_body";
+    codes?: number[];
+    remainingHeader?: string;
+    resetHeader?: string;
+    bodyPath?: string;
+}
+
+export interface PaginationPolicy {
+    strategy: "cursor" | "offset" | "link_header" | "none";
+    cursorParam?: string;
+    cursorPath?: string;
+    offsetParam?: string;
+    limitParam?: string;
+    linkHeaderName?: string;
+    pageSizeParam?: string;
+    defaultPageSize?: number;
+}
+
+export interface RetryPolicy {
+    retryOn5xx?: boolean;
+    retryOn429?: boolean;
+    customRetryCondition?: (error: {
+        status?: number;
+        message?: string;
+        response?: unknown;
+    }) => boolean;
+    maxRetries?: number;
+    backoffStrategy?: "exponential" | "linear" | "fixed";
+    initialDelay?: number;
+}
+
+export interface ErrorPolicy {
+    classifyError?: (error: {
+        status?: number;
+        message?: string;
+    }) => "user" | "system" | "unknown";
+    userErrorCodes?: number[];
+    systemErrorCodes?: number[];
+}
+
+export interface StreamingPolicy {
+    enabled: boolean;
+    mechanism?: "sse" | "chunked" | "websocket";
+    parser?: "ndjson" | "json" | "text" | "custom";
+}
+
+export interface ConnectorPolicies {
+    rateLimit?: RateLimitPolicy;
+    pagination?: PaginationPolicy;
+    retry?: RetryPolicy;
+    error?: ErrorPolicy;
+    streaming?: StreamingPolicy;
+}
+
+export interface ActionDefinition<TInput = unknown, TOutput = unknown> {
+    isTool?: boolean;
+    input: z.ZodSchema<TInput>;
+    output?: z.ZodSchema<TOutput>;
```

Comment:
It would be great to also have an error schema, which can be a union of responses derived from OpenAPI or it could be a part of output.

Reply:
added! thanks